File: LeetCode/Python/0844_Backspace_String_Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        s_stack = []
        t_stack = []
        
        for char in s:
            if char == "#":
                if len(s_stack):
                    s_stack.pop()
            else:
                s_stack.append(char)
                
        for char in t:
            if char == "#":
                if len(t_stack):
                    t_stack.pop()
            else:
                t_stack.append(char)
                
        return s_stack == t_stack

File: LeetCode/Java/0844. Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        int len;
        char c;
        
        StringBuilder sFinal = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            c = s.charAt(i);
            len = sFinal.length();
            
            if (c == '#') {
                if (len != 0)
                    sFinal.deleteCharAt(len - 1);
            } else
                sFinal.append(c);
        }
        
        StringBuilder tFinal = new StringBuilder();
        for (int i = 0; i < t.length(); i++) {
            c = t.charAt(i);
            len = tFinal.length();
            
            if (c == '#') {
                if (len != 0)
                    tFinal.deleteCharAt(len - 1);
            } else
                tFinal.append(c);
        }

        return tFinal.toString().equals(sFinal.toString());
    }
}
